src/main/java/Duke/task/TaskList.java

```diff
@@ -0,0 +1,94 @@
+package Duke.task;
+
+import Duke.model.Deadline;
+import Duke.model.Event;
+import Duke.model.ToDo;
+import Duke.utils.FileHelper;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskList {
+    private List<Task> tasks;
+    private String path;
+    private String fileName;
+
+    public TaskList(String path, String fileName) {
+        tasks = new ArrayList<>();
+        this.path = path;
+        this.fileName = fileName;
+        loadDataFromFile();
+    }
+
+    private void loadDataFromFile() {
+        List<String> data = FileHelper.ReadFromFile(path, fileName);
+        for(int i = 0; i < data.size(); i++) {
+            addTask(data.get(i));
+        }
+    }
+
+    public void save(){
+        FileHelper.save(path, fileName, getStringifiedList());
+    }
+
+
+    public void addTask(String data) {
+        Task task;
+        String[] parts = data.split("( \\| )");
+        switch(parts[0]){
+            case("T"):
```

Comment:
Should there be no indentation after switch case? You can change the IDE formatting in settings for this :)

Reply:
I notice this is some other places as well. Feel free to edit